src/lib/model_selection/_split.ts

```diff
@@ -208,3 +210,188 @@ export function train_test_split(
     yTrain: clean(yTrain),
   };
 }
+
+const rangeValidationError = (type, size, n_samples) => `${type}=${size} should be either 
+positive and smaller than number of samples ${n_samples} or a float in (0, 1) range`;
+
+const testRangeValidationError = (test_size, n_samples) => rangeValidationError('test_size', test_size, n_samples);
+
+const trainRangeValidationError = (test_size, n_samples) => rangeValidationError('test_size', test_size, n_samples);
+
+function uniq(array: Type1DMatrix<any> = null): Type2DMatrix<number> {
+  const uniqArray = _.uniq(_.flatten(array)).sort();
+  let min = Number.MAX_VALUE;
+  let max = Number.MIN_VALUE;
+
+  const valueCount = {};
+  const uniqIndexMap = uniqArray.reduce((acc, ele, i) => {
+    if (min > ele) {
+      min = ele;
+    }
+
+    if (max < ele) {
+      max = ele;
+    }
+
+    return {
+      ...acc,
+      [ele]: i,
+    };
+  }, {});
+
+  const indexMap = array.map((ele) => {
+    if (valueCount[ele]) {
+      valueCount[ele] += 1;
+    } else {
+      valueCount[ele] = 1;
+    }
+    return uniqIndexMap[ele];
+  });
+
+  return [uniqArray, indexMap];
+}
+
+function binCount(array: number[]) {
+  const min: number = _.min(array);
+  const max: number = _.max(array);
+
+  const arrToObj = array.reduce(
+    (acc, ele, i) => ({
+      ...acc,
+      [ele]: i,
+    }),
+    {},
+  );
+  const retArray = Array(max - min).fill(0);
+  for (let i = 0; i < retArray.length; i++) {
+    if (arrToObj[i + min]) {
+      retArray[i] = arrToObj[i + min];
+    }
+  }
+
+  return retArray;
+}
+export class StratifiedShuffleSplit {
+  // private n_splits: number;
+  private test_size: number;
+  private train_size: number;
+  // private random_state: number;
+  private default_test_size: number = 0.1;
+  constructor(
+    // n_splits: number = 10,
+    test_size: number = null,
+    train_size: number = null,
+    // random_state: number = null
+  ) {
+    // this.n_splits = n_splits;
+    this.test_size = test_size;
+    this.train_size = train_size;
+    // this.random_state = random_state;
+  }
+
+  split = (X: Type1DMatrix<any> | Type2DMatrix<any> = null, y: Type1DMatrix<any> = null): any[] => {
+    const XTensor = convertToTensor(X);
+    // const yTensor = convertToTensor(y);
```

Comment:
Please remove the comment =)

Reply:
the commented code is work in progress, unused variable, because git pre commit hook wont let me commit with unused variables, but they will be removed soon